src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -97,15 +99,8 @@ private Mono<ShippingInstructionTO> getDeepObjectsForShippingInstruction(
             shippingInstructionRepository
                 .findCarrierBookingReferenceByShippingInstructionID(shippingInstruction.getId())
                 .collectList()
-                .doOnNext(
-                    cRefs -> {
-                      // we should only set carrier booking reference on root SI TO if
-                      // there is only one distinct value
-                      // This is as per requirement
-                      if (cRefs.size() == 1) {
-                        siTO.setCarrierBookingReference(cRefs.get(0));
-                      }
-                    }),
+                .filter(strings -> strings.size() == 1)
+                .doOnNext(cRefs -> siTO.setCarrierBookingReference(cRefs.get(0))),
```

Comment:
This is not the same!

This always hoist a value up to the root level - but we should only do that if all the values are identical!

Reply:
nvm, misread.